app/main.py

```diff
@@ -1,3 +1,50 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, distance: int | float) -> None:
+        self.km = distance
+
+    @staticmethod
+    def is_int(other: int | float | Distance) -> bool:
+        return isinstance(other, (int, float))
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if self.is_int(other):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if self.is_int(other):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __eq__(self, other: int | float | Distance) -> bool:
+        return self.km == other if self.is_int(other) else self.km == other.km
+
+    def __gt__(self, other: int | float | Distance) -> bool:
+        return self.km > other if self.is_int(other) else self.km > other.km
+
+    def __ge__(self, other: int | float | Distance) -> bool:
+        return self.km >= other if self.is_int(other) else self.km >= other.km
+
+    def __le__(self, other: Distance | int | float) -> bool:
+        return not self.__gt__(other)
```

Comment:
it's not a good way to call magic methods, you don't add numbers using `__add__`, right?
it's better to use return `not self > other`

Reply:
In python u need only implement 2 equality methods for comparison functioning in normal(pythonic) way, but since we implemented Distance, which can't be <= 
![image](https://user-images.githubusercontent.com/107141441/217613232-f00d1295-661d-4018-b264-67c394da58d8.png)
it's exact default way for python, when function which checks something is overloaded, and u cant implement easy way checking it by not this method. But to pass tests I needed to implement those methods(since we invented class Distance), so `not other dunder` is great solution as for me.

Reply:
By the way I still disagree on this exact case. I mean since we already overloaded built-in magic for defining our own magic, it is pretty exact logic of python. I mean mundane | magic.
Really seems more logical, it's like we try to turn steering wheel, it doesn't work the way we want, and after replacing it on other wheel, we still hang out our old wheel on this wheel. If you understand the idea.
If the implementation is easy to explain, it may be a good idea )

Reply:
https://www.tutorialsteacher.com/python/magic-methods-in-python
you can read this
btw, the last question, do you use `int.__str__(1)` or `str(1)`?

Reply:
> https://www.tutorialsteacher.com/python/magic-methods-in-python you can read this btw, the last question, do you use `int.__str__(1)` or `str(1)`?

mostly I use __str__ if I want to see class logic which is more defined, or print(1) if it's builtin
I read that, and idc, there's no much suggestion either we should use magic methods inside or not, only suggestions to implement override if we need it:
![image](https://user-images.githubusercontent.com/107141441/217776189-22fed4cc-81b0-43cd-b762-f82ba3053e89.png)

As for me, it's some efficient encapsulation, like when we use lq, in core python gets to method, reads logic, after that goes to dunder method and applies him, while if we introduce that logic inside dunder, python will just use other defined dubder, which seems more efficient to me...
That's my overall idea in using functions, like if we don't need magic methods - better to use basic logic, but if we need magic, use other magic inside magic for encapsulation and efficiency.